### PR TITLE
Switch from eslintrc.json to es.config.js

### LINKED ISSUE #2472: Switching from eslintrc.json to eslint.config.json
**Describe the bug**
We need to switch from `eslintrc.json` to `eslint.config.js` as the former is considered deprecated now.

**To Reproduce**

1. Make changes in any file in the code base. 
2. Add all the changes using `git add .` command
3. Try and commit the changes using `git commit -m "<Your commit message>"`
4. Examine the error, The process fails after the initial few tests.

**Expected behavior**
The changes should have been committed without any such error messages.

**Actual behavior**
After the initial few tasks the commit action fails. Since "`.eslintrc.json`" file is no longer supported. And it asks to switch over to `eslint.config.js` file.
More over the property `.eslintignore` is also not supported rather user is supposed to use the "ignores" property in `eslint.config.js`.

**Screenshots**
image

**Additional details**
None.

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Suyash878**
**What kind of change does this PR introduce?**
Bug fix

**Issue Number:**
Fixes #2472

**Did you add tests for your changes?**
Not relevant

**Snapshots/Videos:**
image

**If relevant, did you update the documentation?**
Not relevant here.

**Summary**
This PR migrates the deprecated version of `eslintrc.json` to `eslint.config.js`.
Now eslint will correctly fix all the linting issues which was not the case earlier as mentioned in the screenshot of the issue raised. 

**Does this PR introduce a breaking change?**
Not sure if relevant here.

**Other information**
None

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

- **New Features**
	- Added a "Videos" section in the `CONTRIBUTING.md` file to enhance resource accessibility for contributors.

- **Bug Fixes**
	- Improved error handling by allowing console logging for unhandled promise rejections, aiding in debugging.

- **Documentation**
	- Minor formatting adjustments in `README.md` to enhance visual structure.
	- Enhanced navigability in the `CONTRIBUTING.md` document.

- **Refactor**
	- Updated ESLint configuration to enforce stricter code quality standards, especially for TypeScript.
	- Modified pre-commit hook to streamline the commit process, though it may affect staging behavior.

- **Chores**
	- Updated various development dependencies in `package.json` to improve linting capabilities and ensure compatibility with modern JavaScript features.
	- Introduced scripts to automate ESLint checks and prevent unauthorized file modifications.

### CODE REVIEW

**Comment by Suyash878** [comment]
After successful merging of this PR, I will go ahead and create a similar issue in the Talawa-Admin repository as well.

**Comment by palisadoes** [comment]
Please fix the conflicting file so we may merge

File: .husky/pre-commit [hunk 1]
```diff
@@ -3,9 +3,6 @@
 # Disable the hooks in CI 
 [ -n "$CI" ] && exit 0
 
-# Change to the current directory
-. "$(dirname -- "$0")/_/husky.sh"
```
Comment (palisadoes) [inline .husky/pre-commit:7]: Why was this removed?

File: README.md [hunk 2]
```diff
@@ -55,3 +55,5 @@ Core features include:
 1. Visit our [YouTube Channel playlists](https://www.youtube.com/@PalisadoesOrganization/playlists) for more insights
    1. The "[Getting Started - Developers](https://www.youtube.com/watch?v=YpBUoHxEeyg&list=PLv50qHwThlJUIzscg9a80a9-HmAlmUdCF&index=1)" videos are extremely helpful for new open source contributors.
 
+
```
Comment (palisadoes) [inline README.md:58]: Your PR has too many unnecessary files that don’t relate to your issue. You can from the PR by running the commands below from the root directory of the repository.

This file is not necessary for the PR

git add -u
git reset HEAD path/to/file
git commit

File: package.json [hunk 3]
```diff
@@ -132,6 +135,7 @@
     "eslint-plugin-import": "^2.29.1",
     "eslint-plugin-tsdoc": "^0.3.0",
     "get-graphql-schema": "^2.1.2",
+    "globals": "^15.9.0",
```
Comment (palisadoes) [inline package.json:138]: Why was this added?

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Please make the suggested changes.
2. Make sure that coderabbit.ai approves your work.

**Comment by Suyash878** [comment]
I will open a new PR, since this one has a lot of unnecessary file changes and conflicts.

**Comment by Suyash878** [comment]
I was working on a new branch for opening a new PR regarding this issue. Now in order to migrate and properly use `eslint.config.js` , I need to install these following packages(mentioned in the screenshot) which then lead to conflicts in the `package-lock.json` file. And due to this same reason the package `globals` was also added.
I would like to know how I shall proceed with this then.
image

**Pull request closed without merging**